src/main/java/org/sagebionetworks/EvaluationUtils.java

```diff
@@ -354,17 +396,23 @@ public static void removeAnnotation(SubmissionStatusModifications statusMods, St
 	}
 
 	public static void applyModifications(final SubmissionStatus submissionStatus, final SubmissionStatusModifications statusMods) {
+
 		for (AnnotationBase annot : statusMods.getAnnotationsToAdd()) {
-			if (annot instanceof StringAnnotation) 
+			if (annot instanceof StringAnnotation) {
 				setAnnotation(submissionStatus, annot.getKey(), ((StringAnnotation) annot).getValue(), annot.getIsPrivate());
-			if (annot instanceof LongAnnotation) 
+			}
+			if (annot instanceof LongAnnotation) {
 				setAnnotation(submissionStatus, annot.getKey(), ((LongAnnotation) annot).getValue(), annot.getIsPrivate());
-			if (annot instanceof DoubleAnnotation) 
+			}
+			if (annot instanceof DoubleAnnotation) {
 				setAnnotation(submissionStatus, annot.getKey(), ((DoubleAnnotation) annot).getValue(), annot.getIsPrivate());
+			}
 		}
-		
+
+		org.sagebionetworks.repo.model.annotation.v2.Annotations annotations = AnnotationsTranslator.translateToAnnotationsV2(submissionStatus.getAnnotations());
```

Comment:
If we move this after line 415, we do not have to implement the removeAnnotation